include/cppkafka/message.h

```diff
@@ -240,12 +240,19 @@ class CPPKAFKA_API MessageTimestamp {
     };
 
     /**
-     * Constructs a timestamp object
+     * Constructs a timestamp object using a 'duration'.
      */
     MessageTimestamp(std::chrono::milliseconds timestamp, TimestampType type);
+    
+    /**
+     * Constructs a timestamp object using a 'time_point'.
+     */
+    template <typename Clock, typename Duration = typename Clock::duration>
+    MessageTimestamp(std::chrono::time_point<Clock, Duration> timestamp, TimestampType type);
```

Comment:
it's not clear to me why it makes sense to have this template constructor be public, since its definition is only available in the .cpp file, and there's no explicit instantiation of it. In fact, I don't see *any* instantiations of this constructor

Reply:
Yeah, the definition should be on the header. It would never work the way it is.

Reply:
Actually, true, why is this constructor even here? Nothing uses it and nothing can construct a timestamp besides cppkafka itself.

Reply:
I put up https://github.com/mfontanini/cppkafka/pull/129 to fix this. I don't think we need that overload at all

Reply:
I can move the def from the .cpp file. I initially did it w/o templates, and then after you said it was ok to templetize based on the clock type I just added the templates but forgot to move it to the header file. My bad.